testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/db/DbConnectionPoolMetricsAssertions.java

```diff
@@ -73,181 +84,163 @@ public DbConnectionPoolMetricsAssertions disableUseTime() {
     }
     if (testWaitTime) {
-      testing.waitAndAssertMetrics(
-          instrumentationName,
-          "db.client.connections.wait_time",
-          metrics ->
-              metrics.anySatisfy(
-                  metric ->
-                      assertThat(metric)
-                          .hasUnit("ms")
-                          .hasDescription(
-                              "The time it took to obtain an open connection from the pool.")
-                          .hasHistogramSatisfying(
-                              histogram ->
-                                  histogram.hasPointsSatisfying(
-                                      point ->
-                                          point.hasAttributes(
-                                              Attributes.of(stringKey("pool.name"), poolName))))));
+      waitAndAssertMetrics(
+          "db.client.connections.wait_time", this::verifyConnectionsWaitTimeMetrics);
     }
     if (testUseTime) {
-      testing.waitAndAssertMetrics(
-          instrumentationName,
-          "db.client.connections.use_time",
-          metrics ->
-              metrics.anySatisfy(
-                  metric ->
-                      assertThat(metric)
-                          .hasUnit("ms")
-                          .hasDescription(
-                              "The time between borrowing a connection and returning it to the pool.")
-                          .hasHistogramSatisfying(
-                              histogram ->
-                                  histogram.hasPointsSatisfying(
-                                      point ->
-                                          point.hasAttributes(
-                                              Attributes.of(stringKey("pool.name"), poolName))))));
+      waitAndAssertMetrics("db.client.connections.use_time", this::verifyConnectionsUseTimeMetrics);
     }
   }
+
+  private void waitAndAssertMetrics(
+      String metricName, Consumer<ListAssert<MetricData>> assertions) {
+    testing.waitAndAssertMetrics(instrumentationName, metricName, assertions);
+  }
+
+  private void verifyConnectionsUseTimeMetrics(ListAssert<MetricData> metrics) {
+    metrics.anySatisfy(
+        metric ->
+            assertThat(metric)
+                .hasUnit("ms")
+                .hasDescription(
+                    "The time between borrowing a connection and returning it to the pool.")
+                .hasHistogramSatisfying(this::verifyHistogramPoolName));
+  }
+
+  private void verifyConnectionsWaitTimeMetrics(ListAssert<MetricData> metrics) {
+    metrics.anySatisfy(this::verifyConnectionsWaitTimeMetric);
+  }
+
+  private void verifyConnectionsWaitTimeMetric(MetricData metric) {
+    assertThat(metric)
+        .hasUnit("ms")
+        .hasDescription("The time it took to obtain an open connection from the pool.")
+        .hasHistogramSatisfying(this::verifyHistogramPoolName);
+  }
```

Comment:
I like these `verify*` methods that take a `MetricData`, but wonder if inlining the rest of the methods back into the tests would allow us to have and eat our cake?

Reply:
Are you suggesting to inline some methods back into `assertConnectionPoolEmitsMetrics`? That could be a reasonable compromise that still makes things _better_ anyway I suppose. Every inline up there introduces an indent level, though. Do you think it's just too flat now or what are you thinking?

My instinct is to take it even further and split this into 8 classes, but that felt a little like overkill.

Reply:
I sent a PR to your branch with a proposal: https://github.com/breedx-splk/opentelemetry-java-instrumentation/pull/6

my thought is by splitting things out too much it makes the tests harder to follow (need to jump around a lot)

I haven't played much with the new test assertions, so would be great to get @anuraaga's thoughts as well